### DISCUSSION TITLE
Is minizip thread safe?

### DISCUSSION THREAD

**Discussion by sp-stefano-lanza** [nmoinvaz/minizip#399] [Q&A] [answered]
By that I mean, can I safely spawn multiple threads, each using minizip concurrently to decompress a zip file (unique per thread) ? Or does minizip have an unprotected, internal global state that would prevent this use case ? 
Thanks a lot for the answer.

**Comment by Coeur** [comment]
Potentially related: #375

**Comment by sp-stefano-lanza** [comment]
That's a different use case. I do not want to use multiple threads to concurrently decompress the same file. I want to run multiple instances of minizip concurrently, each operating on a different file.

**Comment by Coeur** [comment, marked as answer]
As far as I know, minizip 2 is designed with a streaming model that avoids any global state usage. So I guess that dealing with separate files in separate folders (to avoid writing conflicts) could work. But don't take my words for granted: try it yourself or wait for Nathan answer.

Note also that some file systems or hardwares aren't happy with writing two files at the same time, because it forces the hardware to frequently jump from one file to another: performances may be better when unzipping one file at a time.

**Comment by nmoinvaz** [comment]
Yes it is.
